File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
// ...

namespace FieldListCustomIcons {
    public partial class Form1 : Form {

        public Form1() {
            InitializeComponent();
            DevExpress.Data.Browsing.Design.ColumnImageProvider.Instance = new CustomColumnImageProvider();
        }

        private void button1_Click(object sender, EventArgs e) {
            XtraReport1 report = new XtraReport1();
            ReportDesignTool designTool = new ReportDesignTool(report);
            designTool.ShowDesignerDialog();
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraReports.UI

' ...
Namespace FieldListCustomIcons

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            DevExpress.Data.Browsing.Design.ColumnImageProvider.Instance = New CustomColumnImageProvider()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim report As XtraReport1 = New XtraReport1()
            Dim designTool As ReportDesignTool = New ReportDesignTool(report)
            designTool.ShowDesignerDialog()
        End Sub
    End Class
End Namespace
